gtsfm/averaging/rotation/shonan.py

```diff
@@ -181,17 +194,36 @@ def run_rotation_averaging(
             return wRi_list
 
         nodes_with_edges = sorted(list(self._nodes_with_edges(i2Ri1_dict, i1Ti2_priors)))
-        old_to_new_idxes = {old_idx: i for i, old_idx in enumerate(nodes_with_edges)}
-
-        between_factors: BetweenFactorPose3s = self.__between_factors_from_2view_relative_rotations(
-            i2Ri1_dict, old_to_new_idxes
-        )
-        between_factors.extend(self._between_factors_from_pose_priors(i1Ti2_priors, old_to_new_idxes))
-
-        wRi_list_subset = self._run_with_consecutive_ordering(
-            num_connected_nodes=len(nodes_with_edges), between_factors=between_factors
-        )
-
+        old_to_new_idxs = {old_idx: i for i, old_idx in enumerate(nodes_with_edges)}
+
+        i2Ri1_dict_remapped = {
+            (old_to_new_idxs[i1], old_to_new_idxs[i2]): i2Ri1 for (i1, i2), i2Ri1 in i2Ri1_dict.items()
+        }
+        num_correspondences_dict: Dict[Tuple[int, int], int] = {
+            (old_to_new_idxs[i1], old_to_new_idxs[i2]): len(v_corr_idxs[(i1, i2)])
+            for (i1, i2) in v_corr_idxs.keys()
+            if (i1, i2) in i2Ri1_dict
+        }
+
+        def _create_factors_and_run() -> List[Rot3]:
+            measurements: gtsam.BinaryMeasurementsRot3 = self.__between_factors_from_2view_relative_rotations(
+                i2Ri1_dict=i2Ri1_dict_remapped, num_correspondences_dict=num_correspondences_dict
+            )
+
+            # between_factors.extend(self._between_factors_from_pose_priors(i1Ti2_priors, old_to_new_idxes))
+
+            wRi_list_subset = self._run_with_consecutive_ordering(
+                num_connected_nodes=len(nodes_with_edges), measurements=measurements
+            )
+            return wRi_list_subset
+
+        try:
+            wRi_list_subset = _create_factors_and_run()
+        except RuntimeError:
+            if self._weight_by_inliers is True:
+                # At times, Shonan's `SparseMinimumEigenValue` fails to compute minimum eigenvalue.
```

Comment:
does this get logged?

Reply:
Added logging now.

Reply:
Should our weighing scheme (1/num_inliers) be changed to something else?